elements/rh-secondary-nav/rh-secondary-nav.ts

```diff
@@ -0,0 +1,389 @@
+   * Closes all open dropdowns before opening specified.
+   * Toggles overlay to open
+   * @param index
+   * @returns {void}
+   */
+  public open(index: number): void {
+    if (index == null) {
+      return;
+    }
+    const dropdown = this.#dropdownByIndex(index);
+    if (dropdown && RhSecondaryNav.isDropdown(dropdown)) {
+      this.close();
+      this.#expand(index);
+      dropdown?.querySelector('a')?.focus();
+      this._overlay?.toggleNavOverlay(this._overlay, true, this);
+    }
+  }
+
+  /**
+   * Public API, closes all open dropdowns
+   * @returns {void}
+   */
+  public close(): void {
+    const dropdowns = this.#allDropdowns();
+    dropdowns.forEach(dropdown => this.#closeDropdown(dropdown));
+  }
+
+  /**
+   * When dropdown event is triggered gets dropdown index that triggered
+   * event then closes all dropdowns.
+   * If the event is to open a dropdown, run #expand(index)
+   * If isMobile is set dispatch an SecondaryNavOverlayEvent event
+   * to open the overlay
+   * @param event - {SecondaryNavDropdownChangeEvent}
+   * @return {void}
+   */
+  @bound
+  private _dropdownChangeHandler(event: SecondaryNavDropdownChangeEvent): void {
+    const index = this.#getDropdownIndex(event.target as Element);
+    if (index === null || index === undefined) {
+      return;
+    }
+    this.close();
+    if (event.expanded) {
+      this.#expand(index);
+    }
+    if (!this._nav?.classList.contains('is-mobile')) {
+      this.dispatchEvent(new SecondaryNavOverlayEvent(event.expanded, event.toggle));
+    }
+  }
+
+  /**
+   * Handles when forcus changes outside of the navigation
+   * If isMobile is set, close the mobileMenu
+   * Closes all dropdowns and toggles overlay to closed
+   * @param event {FocusEvent}
+   */
+  @bound
+  private _focusOutHandler(event: FocusEvent) {
+    const target = event.relatedTarget as HTMLElement;
+    if (target && !target.closest('rh-secondary-nav')) {
+      if (this._isMobile) {
+        this.#closeMobileMenu();
+      }
+      this.close();
+      this._overlay?.toggleNavOverlay(event.target as HTMLElement, false, this);
+    }
+  }
+
+  /**
+   * Handles when the overlay recieves a click event
+   * Closes all dropdowns and toggles overlay to closed
+   * If isMobile then closes mobile menu to closed
+   * @param event {PointerEvent}
+   */
+  @bound
+  private _overlayClickHandler(event: PointerEvent) {
+    this.close();
+    this._overlay?.toggleNavOverlay(event.target as HTMLElement, false, this);
+    if (this._isMobile) {
+      this.#closeMobileMenu();
+    }
+  }
+
+  /**
+   * When isMobile value is changed
+   * Get all open navMenus
+   * If isMobile is true, open mobile menu
+   * If isMobile is false, close mobile menu and close overlay
+   * @param oldVal {boolean | undefined}
+   * @param newVal {boolean | undefined}
+   * @returns {void}
+   */
+  private __isMobileChanged(oldVal?: boolean | undefined, newVal?: boolean | undefined): void {
```

Comment:
perhaps best handled with a (new) effects callback on screensize controller

Reply:
Fixed in c03d4820dbd8a6b95ee2520b3a7f5c4968b6e766